components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/ConfigurationFacade.java

```diff
@@ -263,28 +263,25 @@ public int getMaxLoginAttemptCount() {
 
     private String buildUrl(String defaultContext, Supplier<String> getValueFromFileBasedConfig) {
 
-        String url;
         if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
-            url = buildTenantQualifiedUrl(defaultContext);
+            return buildAbsoluteUrl(defaultContext);
         } else {
-            String authenticationEndpointURLInFile = getValueFromFileBasedConfig.get();
-            if (StringUtils.isNotBlank(authenticationEndpointURLInFile)) {
-                // Use the value configured in the file.
-                url = authenticationEndpointURLInFile;
+            String urlFromFileBasedConfig = getValueFromFileBasedConfig.get();
+            if (StringUtils.isNotBlank(urlFromFileBasedConfig)) {
+                // If the file based URL is set, then we have to return the file based URL.
+                return urlFromFileBasedConfig;
             } else {
-                // Use the default context.
-                url = defaultContext;
+                return defaultContext;
```

Comment:
Should we return the default context here.
Shouldn't that be a absolute public ?

Reply:
Though of building the absolute public at the consumer level to avoid behaviour change for code that consumes this API in the legacy mode